baybe/constraints/conditions.py

```diff
@@ -41,15 +42,36 @@ def _is_not_close(x: ArrayLike, y: ArrayLike, rtol: float, atol: float) -> np.nd
         given tolerances.
 
     """
-    return np.logical_not(np.isclose(x, y, rtol=rtol, atol=atol))
+    return np.logical_not(isclose(x, y, rtol=rtol, atol=atol))
+
+
+def isclose(x: ArrayLike, y: ArrayLike, rtol: float, atol: float) -> np.ndarray:
```

Comment:
1. Can we make this private (since I also saw that a built-in version is on the way)
2. To match our naming convention, should be probably rather called `_is_close`
3. I guess the return type hint is not correct here?

Reply:
fixed 1 and 2, why do you think 4 is not right?

Reply:
Because you enter `ArrayLike`s as input. For instance, when you use the function in polars context, there will be no `nd.ndarray` created anywhere!?

Reply:
not sure, the application of `np.abs` below will not turn stuff into `ndarray` ?

Reply:
Oh, I didn't actually look at the function body in terms of types. Yes, that would give you an `ndarray`, so the type hint is then OK, but now I wonder: how does this interplay with polars? Do they inherently support numpy types?

Reply:
seems so, maybe @rjavadi can clarify but the tests and mypy seem fine

Reply:
No surprise that mypy is happy because we tell it to ignore interaction with `polars` 😄 but good that the tests seem to pass. Let's wait to hear @rjavadi thoughts, then we can close

Reply:
answer from @rjavadi might have gone lost somehow but its [here](https://github.com/emdgroup/baybe/pull/252#issuecomment-2211001064)
resolving